docs/igvc2018/EKFTesting.md

```diff
@@ -0,0 +1,46 @@
+# Localisation Testing
+Documents how we tested for the EKF for IGVC 2018
+
+# Installation and Setup (Software)
+
+1. Run 'cd ~/Snowflake && git checkout master'
+2. Run 'cd ~/Snowflake/setup_scripts && ./setup_udev_rules.sh'
+  - If the script fails make sure that ./install_dependencies.sh successfully completed before.
+3. Compile everything by running 'catkin_make'.
+
+# Installation and Setup (Hardware)
+
+1. Plug the encoder and IMU into the USB hub.
+2. Place your laptop on the tower platform.
+3. Plug the USB hub into your laptop.
+4. Plug the GPS into your laptop on a separate port from the USB hub.
+5. Place the GPS antenna at the peak of the tower.
+6. Move the test cart to an area with good GPS signal
+
+# Collecting rosbags for testing
+
+1. Open up 4 terminals (or equivalent).
+2. Run 'roscore' on the first terminal.
+2. Run 'cd ~/Snowflake && source devel/setup.sh' on the second terminal.
+3. Launch the localisation stack by running 'roslaunch localisation_igvc localisation.launch'.
+  - This will launch the EKF, GPS, and IMU.
+4. Check that GPS has fix by running 'rostopic echo /gps/navsatfix' on the third terminal.
+  - If there's an output and the header is constantly changing, then the GPS has a fix.
+  - Make sure the GPS is on it's own port in the laptop to ensure the best fix.

```

Comment:
just curious, what does it mean by own port?

Reply:
USB Port, i.e. not sharing through a hub.